Sources/WalletConnectModal/Modal/ModalViewModel.swift

```diff
@@ -164,79 +170,104 @@ final class ModalViewModel: ObservableObject {
+            guard let rhs = $1.order else {
                 return true
             }
             
-            return lhsLastTimeUsed > rhsLastTimeUsed
+            return lhs < rhs
         }
     }
     
-    func loadRecentWallets() {
-        RecentWalletsStorage().recentWallets.forEach { wallet in
+    func sortByInstalled() -> [Listing] {
+        self.sorted { lhs, rhs in
+            if lhs.installed, !rhs.installed {
+                return true
+            }
             
-            guard let lastTimeUsed = wallet.lastTimeUsed else {
-                return
+            if !lhs.installed, rhs.installed {
+                return false
             }
             
-            // Consider Recent only for 3 days
-            if abs(lastTimeUsed.timeIntervalSinceNow) > (24 * 60 * 60 * 3) {
-                return
+            return false
+        }
+    }
+    
+    func sortByRecent() -> [Listing] {
+        self.sorted { lhs, rhs in
+            guard let lhsLastTimeUsed = lhs.lastTimeUsed else {
+                return false
             }
             
-            setLastTimeUsed(wallet.id, date: lastTimeUsed)
+            guard let rhsLastTimeUsed = rhs.lastTimeUsed else {
+                return true
+            }
+            
+            return lhsLastTimeUsed > rhsLastTimeUsed
         }
     }
     
-    func saveRecentWallets() {
-        RecentWalletsStorage().recentWallets = Array(wallets.filter {
-            $0.lastTimeUsed != nil
-        }.prefix(5))
+    func filter(searchTerm: String) -> [Listing] {
+        if searchTerm.isEmpty { return self }
+        
+        return self.filter {
+            $0.name.lowercased().contains(searchTerm.lowercased())
+        }
     }
+}
+
+// MARK: - Recent & Installed Wallets
+
+private extension ModalViewModel {
     
-    func setLastTimeUsed(_ walletId: String, date: Date = Date()) {
-        guard let index = wallets.firstIndex(where: {
-            $0.id == walletId
-        }) else {
+    func checkInstalledWallets() {
+        
+        guard let schemes = Bundle.main.object(forInfoDictionaryKey: "LSApplicationQueriesSchemes") as? [String] else {
             return
         }
         
-        var copy = wallets[index]
-        copy.lastTimeUsed = date
-        wallets[index] = copy
-        
-        saveRecentWallets()
+        wallets.forEach {
+            if
+                let walletScheme = $0.mobile.native,
+                !walletScheme.isEmpty,
+                schemes.contains(walletScheme.replacingOccurrences(of: "://", with: ""))
+            {
+                $0.installed = uiApplicationWrapper.canOpenURL(URL(string: walletScheme)!)
+            }
+        }
+    }
+    
+    func loadRecentWallets() {
+        recentWalletStorage.recentWallets.forEach { wallet in
+            guard let lastTimeUsed = wallet.lastTimeUsed else { return }
+            setLastTimeUsed(wallet.id, date: lastTimeUsed)
```

Comment:
we already have wallet instance here, why we need to search wallet by id in setLastTimeUsed again?

Reply:
The wallet instance here is the one loaded from UserDefaults, while the one I'm matching in `setLastTimeUsed` is part of my `wallets` state property.